File: CS50/Problem_2/readability.c
#include <ctype.h>
#include <math.h>
#include <stdio.h>

int main(void)
{

    const int init_length = 1000;
    char init_story[init_length];

    char story[init_length];
    int total_letter = 0;
    int total_words = 0;
    int total_sentences = 0;
    float index = 0;

    for (int i = 0; i <= init_length; i++)
    {
        init_story[i] = ' ';
    }
    // Get the Text
    printf("Text: ");
    fgets(init_story, init_length, stdin);
    for (int i = 0; i <= init_length; i++)
    {
        story[i] = tolower(init_story[i]);
    }
    // Calculate total letters of the story
    for (int i = 0; i <= init_length; i++)
    {
        if (story[i] >= 'a' && story[i] <= 'z')
            total_letter += 1;
    }
    // Calculate number of word in the story
    for (int i = 0; i < init_length; i++)
    {
        if ((story[i] >= 'a' && story[i] <= 'z') && (story[i + 1] < 'a' || story[i + 1] > 'z') &&
            (story[i + 1] != '\'' && story[i + 1] != '-'))
            total_words += 1;
    }
    // Calculate number of sentence in the story
    for (int i = 0; i < init_length; i++)
    {
        if (story[i] == '.' || story[i] == '?' || story[i] == '!')
            total_sentences += 1;
    }
    // Apply Coleman-Liau index
    index = 0.0588 * 100 * total_letter / total_words -
            0.296 * 100 * total_sentences / total_words - 15.8;
    // Print the result
    printf("\ntotal letters: %i\n", total_letter);
    printf("total words: %i\n", total_words);
    printf("total sentences: %i\n", total_sentences);
    if (index < 1)
        printf("Before Grade 1\n");
    else if (index > 15)
        printf("Grade 16+\n");
    else
        printf("Grade %.0f\n", index);
}

File: CS50/Problem_6/sentimental-readability/readability.py
def main():
    text = input("Text: ")
    # Index i, Letters: l, Words: w, Sentences: s, Level: level
    i, l, w, s = 0, 0, 0, 0
    text_len = len(text)

    while (i < text_len):
        # Find word that's not end of the sentence
        if (text[i] == " "):
            if (text[i - 1] != "." and text[i - 1] != "?"):
                w += 1
        # Find letter
        elif (text[i].isalpha()):
            l += 1
        # Find end of the sentence
        elif (text[i] == "?" or text[i] == "."):
            w += 1
            s += 1
        i += 1
    L = l / w * 100
    S = s / w * 100
    level = 0.0588 * L - 0.296 * S - 15.8
    # Print out grade
    if (level < 0.5):
        print("Before Grade 1")
    elif (level < 1.5):
        print("Grade 1")
    elif (level < 2.5):
        print("Grade 2")
    elif (level < 3.5):
        print("Grade 3")
    elif (level < 4.5):
        print("Grade 4")
    elif (level < 5.5):
        print("Grade 5")
    elif (level < 6.5):
        print("Grade 6")
    elif (level < 7.5):
        print("Grade 7")
    elif (level < 8.5):
        print("Grade 8")
    elif (level < 9.5):
        print("Grade 9")
    elif (level < 10.5):
        print("Grade 10")
    else:
        print("Grade 16+")


main()
